data-migration/2.x-to-3.x/harvest_project_info_nlesc.py

```diff
@@ -159,13 +162,63 @@ def get_project_related_projects(soup):
     return project_related_projects
 
 
-def get_project_image_url(soup):
-    for figure in soup.findAll("figure"):
-        for img in figure.findAll("img"):
-            imageUrl = img.get("src")
-            if imageUrl != "":
-                return make_url_secure(imageUrl)
-    return "https://FIXME"
+def get_project_image(soup, maxwidth=1024, maxheight=600):
+    def crop_to_aspect_ratio():
+        target_ar = maxwidth / maxheight
+        actual_ar = img.width / img.height
+        if actual_ar < target_ar:
+            # too narrow, adjust height
+            target_height = math.floor(img.width / target_ar)
+            left = 0
+            upper = math.floor((img.height - target_height) / 2)
+            right = img.width
+            lower = upper + target_height
+            box = (left, upper, right, lower)
+            return img.crop(box)
+        else:
+            # too wide, adjust width
+            target_width = math.floor(img.height * target_ar)
+            left = math.floor((img.width - target_width) / 2)
+            upper = 0
+            right = left + target_width
+            lower = img.height
+            box = (left, upper, right, lower)
+            return img.crop(box)
+
+    def resize():
+        return img.resize((maxwidth, maxheight))
+
+    figures = soup.find_all("figure")
+    if len(figures) == 0:
+        print("Doenst have a <figure> tag, returning dummy data")
+        return {"data": "", "mimeType": "image/jpg"}
+
+    src = figures[0].find("img").attrs.get("data-full-url", None)
+    if src is None:
+        print("<img> didnt have a data-full-url, using src instead")
+        src = figures[0].find("img").attrs.get("src", None)
+        if src is None:
+            print("<img> didnt have an src either, returning dummy data instead")
+            return {"data": "", "mimeType": "image/jpg"}
+
+    response = requests.get(src)
+    if not response.ok:
+        print("Something went wrong retrieving the image from src='{0}'".format(src))
+
+    img_bytes = io.BytesIO(response.content)
+
+    img = Image.open(img_bytes)
+    fmt = img.format
+    img = crop_to_aspect_ratio()
```

Comment:
The crop seems dangerous to me, a very large image could be cropped too much. 
We should check that the images are not cropped beyond recognition.

On the other hand this reduces the size.

Reply:
`crop_to_aspect_ratio` retains either the original width or the original height, so I don't expect this to be a problem.
- if the image is too wide, `crop_to_aspect_ratio` cut off equal parts from the left and right sides
- if the image is too tall, `crop_to_aspect_ratio` cut off equal parts from the top and bottom sides

Reply:
Did not look into the code, thanks for the clarification.
Looking at the thumbnails all project images look OK.